src/main/java/meteordevelopment/meteorclient/systems/modules/world/AutoSmelter.java

```diff
@@ -6,137 +6,125 @@
 package meteordevelopment.meteorclient.systems.modules.world;
 
 import meteordevelopment.meteorclient.mixin.AbstractFurnaceScreenHandlerAccessor;
+import meteordevelopment.meteorclient.settings.BoolSetting;
+import meteordevelopment.meteorclient.settings.ItemListSetting;
+import meteordevelopment.meteorclient.settings.Setting;
+import meteordevelopment.meteorclient.settings.SettingGroup;
 import meteordevelopment.meteorclient.systems.modules.Categories;
 import meteordevelopment.meteorclient.systems.modules.Module;
 import meteordevelopment.meteorclient.utils.player.InvUtils;
+import net.minecraft.block.entity.AbstractFurnaceBlockEntity;
+import net.minecraft.inventory.SimpleInventory;
+import net.minecraft.item.Item;
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.Items;
+import net.minecraft.recipe.RecipeType;
 import net.minecraft.screen.AbstractFurnaceScreenHandler;
 
+import java.util.Arrays;
+import java.util.List;
+
 public class AutoSmelter extends Module {
-    private int step;
-    private boolean first;
-    private int timer;
-    private boolean waitingForItemsToSmelt;
+    private final SettingGroup sgGeneral = settings.getDefaultGroup();
+
+    private final Setting<List<Item>> fuelItems = sgGeneral.add(new ItemListSetting.Builder()
+        .name("fuel-items")
+        .description("Items to use as fuel")
+        .defaultValue(Arrays.asList(Items.COAL, Items.CHARCOAL))
+        .filter(x -> AbstractFurnaceBlockEntity.canUseAsFuel(x.getDefaultStack()))
+        .build()
+    );
+
+    private final Setting<List<Item>> smeltableItems = sgGeneral.add(new ItemListSetting.Builder()
+        .name("smeltable-items")
+        .description("Items to smelt")
+        .defaultValue(Arrays.asList(Items.IRON_ORE, Items.GOLD_ORE, Items.COPPER_ORE, Items.RAW_IRON, Items.RAW_COPPER, Items.RAW_GOLD))
+        .filter(x -> mc.world.getRecipeManager() == null || mc.world.getRecipeManager().getFirstMatch(RecipeType.SMELTING, new SimpleInventory(x.getDefaultStack()), mc.world).isPresent())
+        .build()
+    );
+
+    private final Setting<Boolean> disableWhenOutOfItems = sgGeneral.add(new BoolSetting.Builder()
+        .name("disable-when-out-of-items")
+        .description("Disable the module when you run out of items")
+        .defaultValue(true)
+        .build()
+    );
+
 
     public AutoSmelter() {
         super(Categories.World, "auto-smelter", "Automatically smelts all items in your inventory that can be smelted.");
```

Comment:
This description should probably be changed

Reply:
I shall change it immediately